utilities/workers/worker_manager.py

```diff
@@ -48,3 +48,21 @@ def join_workers(self) -> None:
         """
         for worker in self.__workers:
             worker.join()
+
+    def are_workers_alive(self) -> bool:
+        """
+        Check if workers are alive.
+
+        Return: True if all workers are alive. False is any 1 worker is not alive.
+        """
+        for worker in self.__workers:
+            if not worker.is_alive():
+                return False
```

Comment:
check if the list is empty as well - if the list is empty, it would also indicate that the thread is dead